ksql-parser/src/main/java/io/confluent/ksql/parser/properties/with/WithClauseProperties.java

```diff
@@ -117,4 +119,40 @@ private static void throwOnUnknownProperty(
           + String.join(",", onlyInProvided));
     }
   }
+
+  public Optional<Character> getValueDelimiter() {
+    final String providedValueDelimiter = getString(CommonCreateConfigs.VALUE_DELIMITER_PROPERTY);
+    if (providedValueDelimiter == null) {
+      return Optional.empty();
+    }
+    if (providedValueDelimiter.isEmpty()) {
```

Comment:
Put a validator on the ConfigDef to handle this, rather than explicit code here.